nixos/modules/services/matrix/synapse.nix

```diff
@@ -158,12 +174,33 @@ in {
         '';
       };
 
+      extras = mkOption {
+        type = types.listOf (types.enum (lib.attrNames pkgs.matrix-synapse.optional-dependencies));
+        default = [
+          # Support internationalized domain names in user-search
+          "user-search"
+          # Provide the JournalHandler for the default log_config
+          "systemd"
+        ];
+        example = literalExpression ''
+          [
+            # Provide statistics about caching memory consumption
+            "cache-memory"
+          ]
+        '';
```

Comment:
The example should repeat the default settings otherwise the default log_config stops working.

Reply:
I think we should document somewhere what all of these keys mean, i.e. that systemd is relevant for the log configuration. Adding some of these things as example is probably just confusing (also, the comments for the defaults which also appear relevant won't end up in the manual).

Reply:
Ideally something upstream maintains, but they only have the comments in the [`pyproject.toml`](https://github.com/matrix-org/synapse/blob/a5fb382a29991c8eafcb8c54cdd8c7aab260c237/pyproject.toml#L249-L273) and then some hints sprinkled across the manual.

Reply:
Considering that the example was updated I guess we can resolve this part. Personally I think that we should just do this extra-effort, at least I can't think of a way to retrieve this information at eval-time without making the manual depend on `pkgs.matrix-synapse` (and perhaps even ifd) and that's actually a useful piece of information, so that's fine for me.